shopfloor_reception/services/reception.py

```diff
@@ -768,25 +763,32 @@ def _align_product_uom_qties(self, move):
         # If move is already done, do not update lines qties
         # if move.state in ("done", "cancel"):
         #     return
-
+        move = line.move_id
         qty_todo = move.product_uom_qty
-        qty_done = sum(move.move_line_ids.mapped("qty_done"))
+        qty_done = 0.0
+        move_uom = move.product_uom
+        for move_line in move.move_line_ids:
+            # Use move's uom
+            qty_done += move_uom._compute_quantity(
+                move_line.qty_done, move_line.product_uom_id, round=False
+            )
         rounding = move.product_id.uom_id.rounding
         compare = float_compare(qty_done, qty_todo, precision_rounding=rounding)
-        if compare < 1:  # If qty done <= qty todo, align qty todo on move lines
+        if compare < 1:  # If qty done < qty todo, align qty todo in the response
```

Comment:
I think we should always compute it. If qty done >= qty todo, you should not return 0 as todo. So I would drop this `if`.

Maybe you could skip `line` in above `qty_done` computation; That would be a `other_lines_qty_done`.
Then below, you don't need to re-add `line.qty_done` to the `remaining_todo`

Reply:
If the `if` case is removed, when the users are processing more than the total quantity, how should the computation work ?